iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/env/RemoveDataNodeHandler.java

```diff
@@ -130,23 +130,40 @@ public void changeDataNodeStatus(
 
     for (Map.Entry<Integer, TSStatus> entry :
         changeDataNodeStatusContext.getResponseMap().entrySet()) {
-      long currentTime = System.nanoTime();
-
       int dataNodeId = entry.getKey();
       NodeStatus nodeStatus = nodeStatusMap.get(dataNodeId);
       RegionStatus regionStatus = RegionStatus.valueOf(nodeStatus.getStatus());
+
+      if (!isSucceed(entry.getValue())) {
+        LOGGER.error(
+            "{}, Failed to change DataNode status, dataNodeId={}, nodeStatus={}",
+            REMOVE_DATANODE_PROCESS,
+            dataNodeId,
+            nodeStatus);
+        continue;
+      }
+
       // Force updating NodeStatus
+      long currentTime = System.nanoTime();
```

Comment:
print nanotime is useless?

Reply:
`updateCurrentStatistics` will choose the latest sample, so I print the time here to compare with the sample in the updateCurrentStatistics to check if it is because of the race of normal hearbeat from DataNode.

Reply:
got it